### PR TITLE
homological product code sec attempt

### PR DESCRIPTION
**Pull request by xzheng16**
I guess there are some bugs in the first attempt.

### CODE REVIEW

**Comment by valbert4** [comment]
FYI You can make updates in the same pull request, so you don't have to submit a new one every time.

**Comment by xzheng16** [comment]
This is a test for leaving comments. I think I have fixed some indent errors and syntex errors.

File: codes/quantum/qubits/homological_product.yml [hunk 1]
```diff
@@ -7,16 +7,36 @@ code_id: homological_product
 physical: qubits
 logical: qubits
 
-name: 'Homological product code'
+name: 'Single-sector Homological product code'
 introduced: '\cite{arXiv:1311.0885}'
 
-description: 'Stub.'
+description: |
```
Comment (valbert4) [inline codes/quantum/qubits/homological_product.yml:13]: STart with a sentence outlining the general properties of the code. "CSS code based on tensor product of ..."

File: codes/quantum/qubits/homological_product.yml [hunk 2]
```diff
@@ -7,16 +7,36 @@ code_id: homological_product
...
+        'In the single sector homological product code, each base code is associated with the chain complex \( C_a \longrightarrow C_a\longrightarrow C_a\) 
+                     such that the boundary operator satisfies \(\partial_{a}^{2}=0\). The image \(im\partial\) form the parity check space and any set of basis for this 
+                     space can be viewed as Z-type stabilizer group generators. The same rule holds for \(im\partial^{T}\), which gives X-type stabilizer generators.
```
Comment (valbert4) [inline codes/quantum/qubits/homological_product.yml:16]: \\(Z\\)-type

File: codes/quantum/qubits/homological_product.yml [hunk 3]
```diff
@@ -7,16 +7,36 @@ code_id: homological_product
...
+                     selected from all possible invertible matrices with uniform distribution and \(\delta_0\) is the canonical form \cite{arXiv:1311.0885}.'
 # Could be expanded to include Tanner graphs
 # Defn of CSS has been changed to accomodate this... In contrast to the original CSS construction there are no restrictions on the form of \(C_1\) and \(C_2\).'
 
-protection: ''
+protection: 'Bit-flip errors and phase errors'
 
 features:
+  rate:
+    -'Given two codes \([[n_a, k_a, d_a, w_a]]\) where \(w_a\) denotes the maximum hamming weight of all rows and columns of \(\partial_a\), the homological product code has parameter \([[n=n_1 n_2, k=k_1 k_2, d\leq d_1 d_2, w\leq w_1+w_2]]\).'
```
Comment (valbert4) [inline codes/quantum/qubits/homological_product.yml:35]: Rate is not a list. Move item 1 to protection field. Merge the rest into a small paragraph for rate field.

File: codes/quantum/qubits/homological_product.yml [hunk 4]
```diff
@@ -32,6 +52,13 @@ features:
 relations:
   parents:
     - code_id: css
-      detail: ''
+      detail: 'By definition.'
```
Comment (valbert4) [inline codes/quantum/qubits/homological_product.yml:55]: No detail needed here. Remove CSS as parent.

File: codes/quantum/qubits/homological_product.yml [hunk 5]
```diff
@@ -32,6 +52,13 @@ features:
...
     - code_id: quantum_ldpc
-      detail: ''
+      detail: 'The idea of homological product of two chain complexes appears frequently in the search for qLDPC codes.'
```
Comment (valbert4) [inline codes/quantum/qubits/homological_product.yml:57]: Remove hypergraph and lifted. Those are already cousins to homological product. Their entries are mentioned in the corresponding code files for those codes.

**Comment by valbert4** [comment]
This completes round 1 for this code. Please upload the rest of your codes, as we will need a round to go through them. Please use one commit for all code files per round. Please make sure checks go through.

**Pull request closed without merging**